### PR TITLE
[CodeStyle][task 21] enable ruff C405 rule in `python/paddle/base`

### PR DESCRIPTION
**Pull request by yuchen202**
### PR types

Others

### PR changes

Others

### Description

https://github.com/PaddlePaddle/Paddle/issues/57367

Task 21 enable C405

### CODE REVIEW

File: python/paddle/base/multiprocess_utils.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def _func_register(function):
             if not callable(function):
                 raise TypeError("%s is not callable object." % (function))
             # check function object whether hash-able
-            set([function])
+            {function}
```
Comment (gouzil) [inline python/paddle/base/multiprocess_utils.py]:
```suggestion

```

**Review by SigureMo — DISMISSED** [review]
LGTMeow 🐾

**Review by SigureMo — APPROVED** [review]
LGTMeow 🐾

**Review by SigureMo — APPROVED** [review]
LGTMeow 🐾

**Pull request merged**